editors/code/src/util.ts

```diff
@@ -86,6 +90,32 @@ function sleep(ms: number) {
     return new Promise(resolve => setTimeout(resolve, ms));
 }
 
+export function waitForCancellation(token: vscode.CancellationToken): Promise<void> {
+    return new Promise(resolve => token.onCancellationRequested(resolve));
+}
+
+export function once<TParams extends any[]>(fn: (...params: TParams) => void): typeof fn {
+    let wasCalled = false;
+    return (...params) => {
+        if (wasCalled) return;
+        wasCalled = true;
+        fn(...params);
+    };
+}
+
+export function cancelWhenReentered(fn: (cts: vscode.CancellationToken) => Promise<void>) {
+    let activeCts: null | vscode.CancellationTokenSource = null;
```

Comment:
*activeCtx

Reply:
It stands for "cancellation token source", not "context", but spells similar ...